xdsl/ir/core.py

```diff
@@ -73,13 +73,19 @@ class MLContext:
     _loaded_dialects: dict[str, Dialect] = field(default_factory=dict)
     _loaded_ops: dict[str, type[Operation]] = field(default_factory=dict)
     _loaded_attrs: dict[str, type[Attribute]] = field(default_factory=dict)
+    _registered_dialects: dict[str, Callable[[], Dialect]] = field(default_factory=dict)
+    """
+    A dictionary of all registered dialects that are not yet loaded. This is used to
+    only load the respective Python files when the dialect is actually used.
+    """
```

Comment:
To me, it is not quite intuitive that a dialect is not registered anymore after loading it. I guess it makes some of the implementation easier, but we can just compute the information (registered dialects - loaded dialects) when we need it.

Reply:
You would prefer not removing them? That's fine for me too!

Reply:
Yes, I think it makes the name align with the actual behavior (in a registered dialect is one that previously has been registered).